Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -468,21 +459,24 @@ namespace AzToolsFramework
         m_bookmarkFileName = bookmarkComponent->GetLocalBookmarksFileName();
     }
 
-    bool LocalViewBookmarkLoader::SaveLocalBookmark(const ViewBookmark& bookmark, ViewBookmarkType bookmarkType)
+    bool LocalViewBookmarkLoader::SaveLocalBookmark(const ViewBookmark& bookmark, const ViewBookmarkType bookmarkType)
     {
         SetupLocalViewBookmarkComponent();
 
-        AZStd::string setRegPath;
-        switch (bookmarkType)
+        const AZStd::string setRegPath = [this, bookmarkType]
         {
-        case ViewBookmarkType::Standard:
-            // note: this function is not used currently and will need to be updated when DefaultViewBookmarkCount is removed
-            setRegPath = BookmarkSetRegPath(m_bookmarkFileName, aznumeric_cast<int>(m_localBookmarks.size()) + 1);
-            break;
-        case ViewBookmarkType::LastKnownLocation:
-            setRegPath = "/" + m_bookmarkFileName + "/LastKnownLocation";
-            break;
-        }
+            switch (bookmarkType)
+            {
+            case ViewBookmarkType::Standard:
+                // note: this function is not currently used and will need to be updated when DefaultViewBookmarkCount is removed
+                return BookmarkSetRegPathAtIndex(m_bookmarkFileName, aznumeric_cast<int>(m_localBookmarks.size()) + 1);
+            case ViewBookmarkType::LastKnownLocation:
+                return "/" + m_bookmarkFileName + "/LastKnownLocation";
+            default:
+                AZ_Assert(false, "Invalid ViewBookmarkType found: %d", bookmarkType);
+                return AZStd::string{};
+            }
+        }();
```

Comment:
Maybe it fits better to move the switch code to a function?

Reply:
I'd rather not move this right now as it's not used elsewhere and it's very similar to before (it's just now `setRegPath` can be `const`) - this code is likely to be updated in future though so I'll definitely consider it then 👍